libs/wingsdk/src/ex/dynamodb-table.ts

```diff
@@ -450,82 +853,133 @@ export interface DynamodbTransactWriteItem {
    * A request to perform a put operation.
    */
   readonly put?: DynamodbTransactWriteItemPutProps;
+
   /**
    * A request to perform a update operation.
    */
-  readonly update?: DynamodbTransactWriteItemUpdateProps;
+  readonly update?: DynamodbTransactWriteItemUpdateOptions;
+
   /**
    * A request to perform a delete operation.
    */
-  readonly delete?: DynamodbTransactWriteItemDeleteProps;
+  readonly delete?: DynamodbTransactWriteItemDeleteOptions;
+
+  /**
+   * A request to perform a condition check operation.
+   */
+  readonly conditionCheck?: DynamodbTransactWriteItemConditionCheckOptions;
 }
 
 /**
- * Properties for `DynamodbTable.transactWriteItems`.
+ * Options for `DynamodbTable.transactWriteItems`.
  */
-export interface DynamodbTransactWriteItemsProps {
+export interface DynamodbTransactWriteItemsOptions {
   /**
    * The write transact items.
    */
   readonly transactItems: DynamodbTransactWriteItem[];
 }
 
+/**
+ * Result for `DynamodbTable.transactWriteItems`.
+ */
+export interface DynamodbTableTransactWriteItemsResult {
+  /**
+   * The capacity units consumed by the operation.
+   */
+  readonly consumedCapacity?: Json;
+
+  /**
+   * Information about item collections, if any, that were affected by the operation.
+   */
+  readonly itemCollectionMetrics?: Json;
+
+  /**
+   * The attribute values as they appeared before the operation, but only if `ReturnValues` is specified as `"ALL_OLD"` in the request.
+   */
+  readonly attributes?: Json;
+}
+
 /**
  * Inflight interface for `DynamodbTable`.
  */
 export interface IDynamodbTableClient {
   /**
    * Put an item into the table.
-   * @param item data to be inserted.
-   * @param props dynamodb PutItem props.
+   * @param options dynamodb PutItem options.
    * @inflight
    */
-  putItem(item: Json, props?: DynamodbTablePutItemProps): Promise<void>;
+  putItem(
+    options: DynamodbTablePutItemOptions
+  ): Promise<DynamodbTablePutItemResult>;
 
   /**
    * Get an item from the table.
-   * @param key key of the item.
-   * @param props dynamodb UpdateItem props.
+   * @param options dynamodb UpdateItem options.
    * @inflight
    */
-  updateItem(key: Json, props?: DynamodbTableUpdateItemProps): Promise<Json>;
+  updateItem(
+    options: DynamodbTableUpdateItemOptions
+  ): Promise<DynamodbTableUpdateItemResult>;
 
   /**
    * Delete an item from the table.
-   * @param key key of the item.
+   * @param options dynamodb DeleteItem options.
    * @inflight
    */
-  deleteItem(key: Json): Promise<void>;
+  deleteItem(
+    options: DynamodbTableDeleteItemOptions
+  ): Promise<DynamodbTableDeleteItemResult>;
 
   /**
    * Get an item from the table.
-   * @param key key of the item.
+   * @param options options for the getItem operation.
    * @inflight
-   * @see https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Scan.html
```

Comment:
Yes, I thought it might be misleading to refer to the official docs because:

- They have different naming conventions
- On some methods, they have a different syntax altogether (eg, `BatchWriteItem` allows writing to different tables)

I think it's worth it discussing whether to add these links or not. Maybe we could add one link per inflight method, instead of one link per options property.